publiq-ui/pub-link.vue

```diff
@@ -1,13 +1,25 @@
 <template>
-  <a :href="href" @click="handleClick"><slot /></a>
+  <component :is="as" v-bind="$attrs" v-on="$listeners" @click="handleClick">
+    <slot />
+  </component>
 </template>
 
 <script>
+  import PubButton from './pub-button';
+
+  export const SupportedTypes = ['a', 'nuxt-link', 'pub-button'];
```

Comment:
I'd remove this validation because from the ui lib's perspective we "cannot" know the definitive list of what other types of link components we'd need to support. So if we'd extract this UI lib it will hinder us.

For example `nuxt-link` is an implementation detail specific to Udb3 / Nuxt. But another project could have `other-framework-link` so to speak. `nuxt-link` is also technically not known by the UI lib and Storybook. We could import it because we happen to have all the code in a single repository, but that's not always going to be the case.

Reply:
Yes but we have to import components like PubButton, so we can't just accept PubAlert, because it isn't imported in pub-link

Reply:
So that was the idea behind the validation

Reply:
I mean I could import the whole components directory but that doesn't really make sense to me

Reply:
Ah, I see. That makes sense! I'm just wondering now if there's no other way to achieve that, for example by registering the component we put in the `:as` globally in the Vue app itself? 🤔

Reply:
For example: https://github.com/nuxt/nuxt.js/issues/421

Reply:
I'm not totally sure what you mean. Registering all components as globally known?

Reply:
No, just `nuxt-link` in the Nuxt app.